source/appModules/eclipse.py

```diff
@@ -17,14 +22,83 @@ def event_valueChange(self):
 		# Therefore, just drop this event.
 		pass
 
+class AutocompletionListView(IAccessible):
+
+	def _get_name(self):
+		return AUTOCOMPLETION_LIST_NAME
+
+class AutocompletionListItem(IAccessible):
+
+	def event_selection(self):
+		# Fixme: I don't know if it is the correct way to do this.
+
+		# I think that you can feel more confortable if the text editor gain the focus after the item becomes hidden.
+		if self.appModule.textEditor == None:
+			self.appModule.textEditor = api.getFocusObject()
+
+		if self.appModule.selectedItem != self:
+			self.appModule.selectedItem = self
+			api.setFocusObject(self)
+			self.reportFocus()
+
+	def script_returnToTextEditor(self, gesture):

```

Comment:
Using a script to catch dismissal of the autocomplete isn't ideal. Is there any event that gets fired when the autocomplete is dismissed? You might have to watch for "hide" events on the autocomplete list. Those are filtered out by default, so you'll need to use eventHandler.requestEvents to request them.

If a script is the only way, I'd suggest returning focus to `api.getDesktopObject().objectWithFocus()`, rather than storing the editor. This will make sure focus goes to the "real focus" even if pressing the key caused it to move somewhere other than the editor for some reason.